### PR TITLE
new phase1-coordinator test drop_contributor_state_file_updated_test

### PR DESCRIPTION
**Pull request by kellpossible**
+ New `phase1-coordinator` test `drop_contributor_state_file_updated_test()`
+ Add some new methods to access data in `Coordinator`,
    `CoordinatorState` and `ParticipantInfo`.
+ Remove redundant double use of `lazy_static` and `once_cell`, replace `lazy_static!` with `Lazy` from `once_cell`.

TODO
- [ ] Currently the test is failing because I was expecting that some of the `verifier`'s completed tasks would be moved to `disposed_tasks` after dropping `contributor1`.
- [ ] Update the description in the comment with the list of tests.

### CODE REVIEW

File: phase1-coordinator/src/tests.rs [hunk 1]
```diff
@@ -1300,6 +1301,386 @@ fn try_lock_blocked_test() -> anyhow::Result<()> {
...
+
+        // Check the coordinator state matches what is in storage.
+        assert_eq!(
+            &verifier_info,
+            coordinator_state
+                .current_verifier_info(&verifier)
+                .unwrap()
+                .expect("expected verifier to have info now that it is a current verifier in round 1")
+        );
+
```
Comment (kellpossible) [inline phase1-coordinator/src/tests.rs:1584]: No check here for `assigned_tasks` and currently it is empty at this stage, is that to be expected?

File: phase1-coordinator/src/tests.rs [hunk 2]
```diff
@@ -1300,6 +1301,386 @@ fn try_lock_blocked_test() -> anyhow::Result<()> {
...
```
Comment (kellpossible) [inline phase1-coordinator/src/tests.rs:1665]: No check here for `assigned_tasks` and currently it is empty at this stage, is that to be expected?

File: phase1-coordinator/src/tests.rs [hunk 3]
```diff
@@ -1300,6 +1301,386 @@ fn try_lock_blocked_test() -> anyhow::Result<()> {
...
+        assert!(verifier_info.pending_tasks().is_empty());
+        assert!(!verifier_info.completed_tasks().is_empty());
+        assert!(verifier_info.disposing_tasks().is_empty());
+        // Some of the tasks that depended on participant1's
+        // contributions have been disposed.
+        assert!(!verifier_info.disposed_tasks().is_empty());
```
Comment (kellpossible) [inline phase1-coordinator/src/tests.rs:1671]: This part of the test is failing, perhaps this is an incorrect assumption?

File: phase1-coordinator/src/tests.rs [hunk 4]
```diff
@@ -1300,6 +1301,386 @@ fn try_lock_blocked_test() -> anyhow::Result<()> {
...
+            .current_contributor_info(&contributor2)
+            .unwrap()
+            .expect("expected contributor2 to have info now that it is a current contributor in round 1");
+
+        // Check the coordinator state matches what is in storage.
+        assert_eq!(
+            &contributor2_info,
+            coordinator_state
+                .current_contributor_info(&contributor2)
+                .unwrap()
```
Comment (ibaryshnikov) [inline phase1-coordinator/src/tests.rs:1408]: you can probably replace it with `.current_contributor_info(&contributor2)?`

File: phase1-coordinator/src/tests.rs (hunk 1 above)
Comment (ibaryshnikov) [reply phase1-coordinator/src/tests.rs:1584]: You can add all the checks which are relevant to this test. I think it's good to both check all the cases and avoid duplication, so the checks may vary from test to test.

File: phase1-coordinator/src/tests.rs (hunk 4 above)
Comment (kellpossible) [reply phase1-coordinator/src/tests.rs:1408]: Thanks, good tip!

File: phase1-coordinator/src/tests.rs (hunk 1 above)
Comment (kellpossible) [reply phase1-coordinator/src/tests.rs:1572]: Thanks, my question was more, are we expecting that the verifier should have assigned tasks at this stage? I was thinking maybe it should, but I wasn't sure, so I didn't commit the check/assertion.

File: phase1-coordinator/src/tests.rs (hunk 3 above)
Comment (kellpossible) [reply phase1-coordinator/src/tests.rs:1654]: @raychu86 @howardwu would we expect that if a participant is dropped, the verifier will also move some of its completed tasks (which depend on that contributor) to the `disposed_tasks` list?

Comment (kellpossible) [reply phase1-coordinator/src/tests.rs:1654]: @raychu86 @howardwu  sorry, bump

File: phase1-coordinator/src/tests.rs (hunk 2 above)
Comment (howardwu) [reply phase1-coordinator/src/tests.rs:1648]: Check contributor_1 is empty, contributor_2 has n/2 tasks, verifier has n + n/2 tasks, backup contributor1 has n tasks

**Comment by apruden2008** [comment]
@kellpossible want to change the reviewer to Howard, and we can go through all of these next week?

**Comment by kellpossible** [comment]
@apruden2008 I need to address one of the comments Howard made with the PR then hopefully it should be ready.

**Comment by apruden2008** [comment]
@kellpossible can you please check on the status of this and provide an update?

**Comment by kellpossible** [comment]
@apruden2008 I still need to address the comment Howard made, and ensure that the test passes correctly. I will try to do that today.

**Pull request closed without merging**